### DISCUSSION TITLE
Drop to normal mode on certain commands

### DISCUSSION THREAD

**Discussion by EricDriussi** [ibhagwan/fzf-lua#1554] [Q&A]
Hi there!
I've read around here and in the docs but I can't figure out how (if possible) to configure the plugin so that a given command (`spell_suggest`, for example) would drop me in normal mode instead of insert.
The intention here is to directly use j/k to select an option instead of manually going to normal mode first.

Just as an example, Telescope pickers have a `initial_mode = "normal"` option that does just this.

Did I miss a similar option?

**Comment by ibhagwan** [comment]
There is no option to start in normal mode, IMHO due to the fact fzf runs in the terminal normal mode is not as a normal neovim buffer, try it out by pressing `<C-\><C-n>`.

If you still find it useful you can maybe use the on_create callback to (don’t recall ATM if this happens before insert mode, or use an autocmd for `Filetype` triggering on `fzf` file type.

**Comment by EricDriussi** [comment]
Thanks for the quick feedback.
I just tried the `on_create` callback, but it does not work, it seems it runs before insert mode.
An `autocmd` would also not work, since I would want some commands to behave like this while others to drop me in insert mode.

I understand I might be alone in this and don't expect the plugin to adapt to my specific needs, but I'll draw my use case in case the interest is shared among some other users:

When looking for a spell suggestion, there is not much of a point in typing the word, because all the suggestions are likely very similar (hopefully I don't misspell 'potato' for 'banana', but for something like 'potto'). This means I either have to go to normal mode or use the arrow keys to select the entry.

When parsing LSP references I'm in a similar position. Typing further reduces the results in the list, which might be useful in some cases, but usually I want to look at all the references one at a time.

I could remap the Tab for this, but in the LSP references case it's a shame to lose the default functionality (or having to re-map it as well), since after visual parsing I often need to open a sub-selection of the results.

Of course none of this makes much sense when finding files or grepping, hence why the `autocmd` doesn't quite cut it for me.

Again, thanks for the help (and for the plugin, it's pretty awesome)!

**Reply by ibhagwan** [reply]
Why don’t you try this:
```vim
:FzfLua spell_suggest keymap.fzf.one=accept
```
And then press the number of the result instead?

This will effectively utilize `fzf --bind one:accept` which automatically accepts if there’s only one match (so you can skip pressing enter), consider this similar to pressing `5j` in normal mode.

**Reply by EricDriussi** [reply]
I must be missing something, running that command produces the same behavior as `:FzfLua spell_suggest`

**Reply by ibhagwan** [reply]
> I must be missing something, running that command produces the same behavior as `:FzfLua spell_suggest`

The behavior only applies when you input something in the query that results in **only one line**, it’s also possible that my suggestion doesn’t work as I’m AFK and couldn’t test it, in this case also try sending the raw args:
```lua
:lua require'fzf-lua'.spell_suggest({ cli_args = '--bind one:accept' }
```

Note that for this to work you have to type enough input in the prompt so that there would be only one result.
